src/app/features/swap-form/reducer.ts

```diff
@@ -0,0 +1,123 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { AMM, Position } from '@voltz-protocol/v1-sdk';
+
+import { formatCurrency, toUSFormat } from '../../../utilities/number';
+import { initialiseCashflowCalculator } from './thunks';
+
+type SliceState = {
+  amm: AMM | undefined;
+  position: Position | undefined;
+  // State of prospective swap
+  prospectiveSwap: {
+    // user-inputted notional amount
+    notionalAmount: string;
+  };
+  // State of cashflow calculator
+  cashflowCalculator: {
+    // user-inputted predicted variable apy
+    predictedApy: string;
+    // cached variable factor from termStart to now
+    variableFactorStartNow: number;
+    // Additional cashflow resulted from prospective swap, form now to termEnd.
+    additionalCashflow: string;
+    // Total cashflow resulted from past and prospective swaps, from termStart to termEnd.
+    totalCashflow: string;
+    // Flag that signals whether cashflow calculator was initialised
+    initialised: boolean;
+    // Flag that signals potential errors in cashflow calculation
+    error: boolean;
+  };
+};
+
+const initialState: SliceState = {
+  amm: undefined,
+  position: undefined,
+  prospectiveSwap: {
+    notionalAmount: '0',
+  },
+  cashflowCalculator: {
+    predictedApy: '0',
+    variableFactorStartNow: 0,
+    additionalCashflow: '+0.00',
+    totalCashflow: '+0.00',
+    initialised: false,
+    error: false,
+  },
+};
+
+export const slice = createSlice({
+  name: 'swapForm',
+  initialState,
+  reducers: {
+    setNotionalAmountAction: (
+      state,
+      {
+        payload: { value },
+      }: PayloadAction<{
+        value: string;
+      }>,
+    ) => {
+      state.prospectiveSwap.notionalAmount = value;
+    },
+    setPredictedApyAction: (
+      state,
+      {
+        payload: { value },
+      }: PayloadAction<{
+        value: string;
+      }>,
+    ) => {
+      state.cashflowCalculator.predictedApy = value;
+      if (parseFloat(toUSFormat(value) as string) < 0) {
+        state.cashflowCalculator.error = true;
+      } else {
+        state.cashflowCalculator.error = false;
+      }
+    },
+    refreshCashflows: (
```

Comment:
best practice is to add 'Action' suffix to signal that this is action. Perhaps rename it to updateCashflowCalculatorAction? WDYT

Reply:
Addressed [6b0d2f3](https://github.com/Voltz-Protocol/voltz-ui/pull/570/commits/6b0d2f3843234382bf7a2334f42b64cb5096adce)